### PR TITLE
[macOS] Add arm64 build

### PR DESCRIPTION
**Pull request by flavioislima**
This adds an arm64-optimized build for macOS.

Note:
ARM packages will show as damaged since they are not signed by an Apple ID certified Developer, so this command is required to make it work:
`xattr -d com.apple.quarantine | mdfind kMDItemCFBundleIdentifier = "*heroic"`

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [ ] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

**Comment by flavioislima** [comment]
Well, reopening this one now since to make it work only a small command is required and the benefits are good because performance on native is a lot better so I think it is ready to review again.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -31,36 +31,36 @@ jobs:
...
         with:
-          node-version: '16'
+          node-version: '18'
       - name: Install modules.
         run: yarn
       - name: Build artifacts.
         run: yarn dist:mac --publish=never
       - name: Upload DMG.
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@v3
```
Comment (arielj) [inline .github/workflows/build.yml:63]: just one question cause I don't know how this works, I see this `Upload DMG.` step that was not split in two like the one in the `build-prs-mac.yml` file that now has `Upload x64.` and `Upload ARM64.` steps.

Should this one be split too?

**Comment by arielj** [comment]
This is for the M1/2 chips, right? I only have access to an intel mac so I can't really test this. Code looks fine to me, I just left one question.

File: .github/workflows/build.yml (hunk 1 above)
Comment (flavioislima) [reply .github/workflows/build.yml:63]: good point

**Review by arielj — APPROVED** [review]
Looks good to me, though I can't test this cause I don't have an M1 chip

**Pull request merged**